.github/workflows/check-licenses.yml

```
@@ -11,22 +11,18 @@ jobs:
       - uses: actions/checkout@v4
         with:
           ref: ${{github.event.pull_request.head.sha}}
-
+        
       - name: Cache multiple paths
         uses: actions/cache@v4
         with:
           path: |
             ~/.cache/go-build
             ~/go/pkg/mod
           key: ${{ runner.os }}-build-${{ hashFiles('**/go.sum', '**/go.mod') }}
-
-      - name: Setup Go
-        uses: actions/setup-go@v5
-        with:
-          go-version-file: "${{ github.workspace }}/go.mod"
-          cache: false
-
-      - run: go version
+        
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
```

I would lock to a particular version, otherwise we would be open to breaking changes or malicious take over of the action repo:
```suggestion
      - uses: DeterminateSystems/nix-installer-action@v13
      - uses: DeterminateSystems/magic-nix-cache-action@v7
```
I manually checked both actions and there are both good and bad news:
- **Good news** is, I saw no transitive action dependencies in any of the two YAMLs for the [installer](https://github.com/DeterminateSystems/nix-installer-action/blob/main/action.yml)or the [cache](https://github.com/DeterminateSystems/magic-nix-cache-action/blob/main/action.yml)
- **Bad news** is, there is a ton of JS code in those `dist/index.js` executables to trust from a 3rd party.

---

so what are you suggesting?

---

for now just pin the actions to use a fixed release tag instead of the main branch.